### PR TITLE
Add CUDA support

### PR DESCRIPTION
**Pull request by Colengms**
A global `supportCuda` is set only on activation.  If the language ID is not registered, all CUDA support should effectively be disabled.  Since the cuda-cpp language ID is being added to VS Code itself, we don't expect the availability of the language ID to change at runtime.

Eventually, we could remove the supportCuda setting, when/if we update the version requirement for VS Code to a version that include the cuda-cpp language ID.

### CODE REVIEW

File: Extension/package.json [hunk 1]
```diff
@@ -2209,13 +2224,23 @@
           "command": "workbench.action.showAllSymbols",
           "group": "other1_navigation@4"
         },
+        {
+          "when": "editorLangId == cuda-cpp",
```
Comment (bobbrow) [inline Extension/package.json]: interesting that we add an object for each languageId.  I think we can do `||` in these `when` blocks, but that can be something we look into for a subsequent PR.

Comment (sean-mcmanus) [reply Extension/package.json]: Can we just try that now and see if it works? I recall I tried a long time ago and it didn't work but may be it works now?

Comment (Colengms) [reply Extension/package.json]: Could we defer that to a future PR, just in case?

Comment (sean-mcmanus) [reply Extension/package.json]: Isn't it easy to check to see if it works? I tried and it works. The keyboard command case needs to be formatted like `editorLangId == 'cpp' && editorTextFocus || editorLangId == 'c' && editorTextFocus` without `()` grouping.

Comment (sean-mcmanus) [reply Extension/package.json]: Oh, did you not change the keyboard switch header case? Is that supposed to work with CUDA? I see an issue at https://github.com/microsoft/vscode-cpptools/issues/4509

Comment (sean-mcmanus) [reply Extension/package.json]: Seems like the keybinding should be updated since the context menus exist.

**Review by sean-mcmanus — CHANGES_REQUESTED** [review]
keybindings change?

**Review by sean-mcmanus — APPROVED** [review]

**Pull request merged**